src/H5Dmpio.c

```diff
@@ -4697,8 +4701,8 @@ H5D__mpio_collective_filtered_chunk_update(H5D_filtered_collective_io_info_t *ch
 
             /* Find the chunk entry according to its chunk index */
             HASH_FIND(hh, chunk_hash_table, &chunk_idx, sizeof(hsize_t), chunk_entry);
-            HDassert(chunk_entry);
-            HDassert(mpi_rank == chunk_entry->new_owner);
+            if (chunk_entry == NULL || mpi_rank != chunk_entry->new_owner)
```

Comment:
I'm not a fan of asserts. They are crude and I don't like the way they cause the process to abort. I also don't want people to waste time arguing over whether something warrants a library error or an assert. I feel like asserts should just be used for things that would cause an abort anyway, like dereferencing a NULL pointer or dividing by zero.

Reply:
The problem I see is that if the new_owner check fails, it's likely going to abort or hang anyway because internal library state is broken. 

At the very least though, I think the new_owner check should at least be in a separate error check to distinguish between not finding a chunk entry and the chunk entry's state being messed up. Giving an error to the user that the state is messed up won't really be helpful to them in any way from their perspective, but I guess it at least won't be misleading.

Reply:
I moved the new_owner check into a separate error check